frontend/src/modules/EditZing/Components/NotesModal.tsx

```diff
@@ -0,0 +1,135 @@
+import { useState, useEffect } from 'react'
+import { useParams } from 'react-router-dom'
+import {
+  Modal,
+  Box,
+  Button,
+  Typography,
+  OutlinedInput,
+  IconButton,
+} from '@mui/material'
+import LoadingButton from '@mui/lab/LoadingButton'
+import SaveIcon from '@mui/icons-material/Save'
+import CloseIcon from '@mui/icons-material/Close'
+
+import { API_ROOT } from '@core/Constants'
+import axios from 'axios'
+
+const NotesModal = (props: any) => {
+  const { courseId } = useParams<{ courseId: string }>()
+  const { open, student, handleClose, setSaved, setNotSaved } = props
+  const [note, setNote] = useState(student.notes)
+  const [saving, setSaving] = useState(false)
+  const email = student.email
+
+  // getting notes for student
+  useEffect(() => {
+    const savedNote = student.groups.find((g: any) => g.courseId === courseId)
```

Comment:
`student.groups` is of type `GroupMembership[]`. This would be solved automatically by typing the props of this component. Also add `notes` and `notesModifyTime` to that type definition

Reply:
Yep fixed the typing too